lib/stdlib/test/property_test/base64_prop.erl

```diff
@@ -0,0 +1,429 @@
+    common_decode_malformed(noisy_b64(), fun base64:mime_decode_to_string/1).
+
+common_decode_noisy(Fn) ->
+    ?FORALL(
+        {_, NoisyB64},
+        ?SUCHTHAT({NormalizedB64, NoisyB64}, noisy_b64(), NormalizedB64 =/= NoisyB64),
+        try
+            Fn(NoisyB64)
+        of
+            _ ->
+                false
+        catch
+            error:_ ->
+                true
+        end
+    ).
+
+common_decode_malformed(Gen, Fn) ->
+    ?FORALL(
+        MalformedB64,
+        ?LET(
+            {{NormalizedB64, NoisyB64}, Malformings},
+            {
+                Gen,
+                oneof(
+                    [
+                        [b64_char()],
+                        [b64_char(), b64_char()],
+                        [b64_char(), b64_char(), b64_char()]
+                    ]
+                )
+            },
+            {NormalizedB64, insert_noise(NoisyB64, Malformings)}
+        ),
+        try
+            Fn(MalformedB64)
+        of
+            _ ->
+                false
+        catch
+            error:_ ->
+                true
+        end
+    ).
+
+%%%%%%%%%%%%%%%%%%
+%%% Generators %%%
+%%%%%%%%%%%%%%%%%%
+
+%% Generate a single character from the base64 alphabet.
+b64_char() ->
+    oneof(
+        lists:seq($0, $9) ++
+        lists:seq($A, $Z) ++
+        lists:seq($a, $z) ++
+        [$+, $/]
+    ).
+
+%% Generate a string of characters from the base64 alphabet,
+%% including padding if needed.
+b64_string() ->
+    ?LET(
+        {L, Filler},
+        {list(b64_char()), b64_char()},
+        case length(L) rem 4 of
+            0 -> L;
+            1 -> L ++ [Filler, $=, $=];
+            2 -> L ++ [$=, $=];
+            3 -> L ++ [$=]
+        end
+    ).
+
+%% Generate a binary of characters from the base64 alphabet,
+%% including padding if needed.
+b64_binary() ->
+    ?LET(
+        L,
+        b64_string(),
+        list_to_binary(L)
+    ).
+
+%% Generate a string or binary of characters from the
+%% base64 alphabet, including padding if needed.
+b64() ->
+    oneof([b64_string(), b64_binary()]).
+
+%% Generate a string or binary of characters from the
+%% base64 alphabet, including padding if needed, with
+%% whitespaces inserted at random indexes.
+wsped_b64() ->
+    ?LET(
+        {B64, Wsps},
+        {b64(), list(oneof([$\t, $\r, $\n, $\s]))},
+        {B64, insert_noise(B64, Wsps)}
+    ).
+
+%% Generate a single character outside of the base64 alphabet.
+%% As whitespaces are allowed but ignored in base64, this generator
+%% will produce no whitespaces, either.
+non_b64_char() ->
+    oneof(
+        lists:seq(16#00, 16#08) ++
+        % HTAB, LF
+        [16#0B, 16#0C] ++
+        % CR
+        lists:seq(16#0E, 16#1F) ++
+        % SP
+        lists:seq(16#21, 16#2A) ++
+        % +
+        lists:seq(16#2C, 16#2E) ++
+        % /, 0-9
+        lists:seq(16#3A, 16#40) ++
+        % A-Z
+        lists:seq(16#5B, 16#60) ++
+        % a-z
+        lists:seq(16#7B, 16#FF)
+    ).
```

Comment:
It is hard to understand whether this code produces the correct list of non-base64 characters.

I suggest defining a helper function called `b64_chars/0` (which can be used also by the `b64_char/0` generator).

```suggestion
    oneof(lists:seq(16#00, 16#ff) -- b64_chars()).
```